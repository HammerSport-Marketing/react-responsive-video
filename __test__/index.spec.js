import React from 'react'
import Video from '../src'
import renderer from 'react-test-renderer'
import { mount, shallow } from 'enzyme'

describe('ReactVideo', () => {
  describe('Tag "video"', () => {
    test('Video component with mp4', () => {
      const component = renderer.create(
        <Video mp4="http://commondatastorage.googleapis.com/gtv-videos-bucket/sample/BigBuckBunny.mp4" />,
      )
      let tree = component.toJSON()
      expect(tree).toMatchSnapshot()
    })
    
    test('Video component with ogg', () => {
      const component = renderer.create(
        <Video ogg="http://techslides.com/demos/sample-videos/small.ogv" />,
      )
      let tree = component.toJSON()
      expect(tree).toMatchSnapshot()
    })
    
    test('Video component with webm', () => {
      const component = renderer.create(
        <Video webm="http://techslides.com/demos/sample-videos/small.webm" />,
      )
      let tree = component.toJSON()
      expect(tree).toMatchSnapshot()
    })
    
    test('Video component with HTTP streaming', () => {
      const component = renderer.create(
        <Video m3u8="https://player.vimeo.com/external/375031697.m3u8?s=326616eb51bb4e8c17bc634da4fd784c2a700c9e" />,
      )
      let tree = component.toJSON()
      expect(tree).toMatchSnapshot()
    })
    
   

    test('Video component with responsive wrapper and mp4', () => {
      //const breakpoints = [40,50,60,70].map(n => n + 'em')
      const component = renderer.create(
        <Video 
          mp4="http://techslides.com/demos/sample-videos/small.mp4"
          height={[400,480,600,760].map(n => n + 'px')}
        />,
      )
      let tree = component.toJSON()
      expect(tree).toMatchSnapshot()
    })
  }),
  describe('Tag "iframe"', () => {
    test('Video component with Vimeo', () => {
      const component = renderer.create(
        <Video vimeo="https://vimeo.com/366333472" />,
      )
      let tree = component.toJSON()
      expect(tree).toMatchSnapshot()
    })
  })
})

describe('Vimeo', () => {
  it('renders correctly', () => {
    const wrapper = mount(<Video vimeo="https://vimeo.com/366333472" />)
    expect(wrapper).toMatchSnapshot()
  })
})

