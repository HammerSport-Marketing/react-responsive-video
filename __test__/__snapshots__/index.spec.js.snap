// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReactResponsiveVideo Tag "iframe" Video component with Vimeo 1`] = `
<div
  className="css-k008qs"
>
  
  <iframe
    allow="autoplay; fullscreen"
    allowFullScreen={true}
    className="css-0"
    frameBorder="0"
    height="auto"
    margin={0}
    src="https://vimeo.com/366333472?autoplay=1&loop=1&muted=1&autopause=0"
    width="100%"
  />
</div>
`;

exports[`ReactResponsiveVideo Tag "video" Video component with HTTP streaming 1`] = `
<div
  className="css-k008qs"
>
  <video
    autoPlay={true}
    className="css-0"
    controls={false}
    loop={true}
    m3u8="https://player.vimeo.com/external/375031697.m3u8?s=326616eb51bb4e8c17bc634da4fd784c2a700c9e"
    mp4=""
    muted={true}
    ogg=""
    playsInline={true}
    poster=""
    preload="true"
    webm=""
    width="100%"
  >
    
    
    
    <source
      className="css-0"
      src="https://player.vimeo.com/external/375031697.m3u8?s=326616eb51bb4e8c17bc634da4fd784c2a700c9e"
      type="application/x-mpegURL"
    />
  </video>
  
</div>
`;

exports[`ReactResponsiveVideo Tag "video" Video component with mp4 1`] = `
<div
  className="css-k008qs"
>
  <video
    autoPlay={true}
    className="css-0"
    controls={false}
    loop={true}
    m3u8=""
    mp4="http://commondatastorage.googleapis.com/gtv-videos-bucket/sample/BigBuckBunny.mp4"
    muted={true}
    ogg=""
    playsInline={true}
    poster=""
    preload="true"
    webm=""
    width="100%"
  >
    <source
      className="css-0"
      src="http://commondatastorage.googleapis.com/gtv-videos-bucket/sample/BigBuckBunny.mp4"
      type="video/mp4"
    />
    
    
    
  </video>
  
</div>
`;

exports[`ReactResponsiveVideo Tag "video" Video component with ogg 1`] = `
<div
  className="css-k008qs"
>
  <video
    autoPlay={true}
    className="css-0"
    controls={false}
    loop={true}
    m3u8=""
    mp4=""
    muted={true}
    ogg="http://techslides.com/demos/sample-videos/small.ogv"
    playsInline={true}
    poster=""
    preload="true"
    webm=""
    width="100%"
  >
    
    <source
      className="css-0"
      src="http://techslides.com/demos/sample-videos/small.ogv"
      type="video/ogg"
    />
    
    
  </video>
  
</div>
`;

exports[`ReactResponsiveVideo Tag "video" Video component with responsive wrapper and mp4 1`] = `
<div
  className="css-mqrn5k"
>
  <video
    autoPlay={true}
    className="css-1oh3jdm"
    controls={false}
    loop={true}
    m3u8=""
    mp4="http://techslides.com/demos/sample-videos/small.mp4"
    muted={true}
    ogg=""
    playsInline={true}
    poster=""
    preload="true"
    webm=""
    width="100%"
  >
    <source
      className="css-0"
      src="http://techslides.com/demos/sample-videos/small.mp4"
      type="video/mp4"
    />
    
    
    
  </video>
  
</div>
`;

exports[`ReactResponsiveVideo Tag "video" Video component with webm 1`] = `
<div
  className="css-k008qs"
>
  <video
    autoPlay={true}
    className="css-0"
    controls={false}
    loop={true}
    m3u8=""
    mp4=""
    muted={true}
    ogg=""
    playsInline={true}
    poster=""
    preload="true"
    webm="http://techslides.com/demos/sample-videos/small.webm"
    width="100%"
  >
    
    
    <source
      className="css-0"
      src="http://techslides.com/demos/sample-videos/small.webm"
      type="video/webm"
    />
    
  </video>
  
</div>
`;

exports[`Vimeo renders correctly 1`] = `ReactWrapper {}`;
